### DISCUSSION TITLE
How can I add delay to requests to prevent going over API limit on search results?

### DISCUSSION THREAD

**Discussion by 82d28a** [censys/censys-python#61] [Q&A] [answered]
Please consider adding a configurable delay for searches with large returns to prevent going over the API limit.  

I was able to get around the problem by inserting time.sleep(0.9) after line 167 in api.py. 

Thanks

**Comment by thehappydinoa** [comment, marked as answer]
Hey! Thank you for this suggestion, we actually have something implemented called backoff which automatically does our rate limiting. Please let me know if you have any more ideas.

**Reply by 82d28a** [reply]
Thank you, I will give it a try.
